2022/day10/day10.rs

```diff
@@ -0,0 +1,95 @@
+//! [Day 10: Cathode-Ray Tube](https://adventofcode.com/2022/day/10)
+
+use clap::Parser;
+
+#[derive(Parser)]
+struct Args {
+    /// Puzzle input
+    #[arg(default_value = "input.txt")]
+    path: String,
+}
+
+struct Puzzle {
+    cycles: Vec<i32>,
+}
+
+impl Puzzle {
+    fn new() -> Self {
+        Self { cycles: vec![] }
+    }
+
+    /// Loads data from input (one line)
+    fn configure(&mut self, path: &str) {
+        let data = std::fs::read_to_string(path).unwrap();
+        let lines = data.split('\n').collect::<Vec<_>>();
+
+        #[allow(non_snake_case)]
+        let mut X = 1;
+        for line in lines {
+            if line == "noop" {
+                self.cycles.push(X);
+            } else if let Some(v) = line.strip_prefix("addx ") {
+                self.cycles.push(X);
+                X += v.parse::<i32>().unwrap();
+                self.cycles.push(X);
+            }
+        }
+    }
+
+    // Solves part one
+    fn part1(&self) -> i32 {
+        let mut signal_strength = 0;
+        for (i, x) in self.cycles.iter().enumerate() {
+            if (i + 2 + 20) % 40 == 0 {
```

Comment:
`+1` because the puzzle counts starting at 1
`+1` becayse the signal strength is computed after a cycle